### PR TITLE
NetworkCloud2024-07-01 Stable API : powershell updates

### PR DESCRIPTION
**Pull request by priyamshet**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request
ChangeLog:

General availability of Az.NetworkCloud module with 2024-07-01 API version
Deploy-AzNetworkCloudCluster
Disable-AzNetworkCloudStorageApplianceRemoteVendo…
Enable-AzNetworkCloudStorageApplianceRemoteVendor…
Get-AzNetworkCloudAgentPool
Get-AzNetworkCloudBareMetalMachine
Get-AzNetworkCloudBareMetalMachineKeySet
Get-AzNetworkCloudBmcKeySet
Get-AzNetworkCloudCluster
Get-AzNetworkCloudClusterManager
Get-AzNetworkCloudConsole
Get-AzNetworkCloudKubernetesCluster
Get-AzNetworkCloudKubernetesClusterFeature
Get-AzNetworkCloudL2Network
Get-AzNetworkCloudL3Network
Get-AzNetworkCloudMetricsConfiguration
Get-AzNetworkCloudRack
Get-AzNetworkCloudRackSku
Get-AzNetworkCloudServicesNetwork
Get-AzNetworkCloudStorageAppliance
Get-AzNetworkCloudTrunkedNetwork
Get-AzNetworkCloudVirtualMachine
Get-AzNetworkCloudVolume
Invoke-AzNetworkCloudBareMetalMachineCordon
Invoke-AzNetworkCloudBareMetalMachineDataExtract
Invoke-AzNetworkCloudBareMetalMachineReimage
Invoke-AzNetworkCloudBareMetalMachineReplace
Invoke-AzNetworkCloudBareMetalMachineRunCommand
Invoke-AzNetworkCloudBareMetalMachineRunReadComma…
Invoke-AzNetworkCloudBareMetalMachineUncordon
Invoke-AzNetworkCloudClusterContinueVersionUpdate
Invoke-AzNetworkCloudClusterVersionUpdate
Invoke-AzNetworkCloudScanClusterRuntime
Invoke-AzNetworkCloudVirtualMachineReimage
New-AzNetworkCloudAgentPool
New-AzNetworkCloudBareMetalMachineConfigurationDa…
New-AzNetworkCloudBareMetalMachineKeySet
New-AzNetworkCloudBgpAdvertisementObject
New-AzNetworkCloudBgpServiceLoadBalancerConfigura…
New-AzNetworkCloudBmcKeySet
New-AzNetworkCloudCluster
New-AzNetworkCloudClusterManager
New-AzNetworkCloudConsole
New-AzNetworkCloudControlPlaneNodeConfigurationOb…
New-AzNetworkCloudEgressEndpointObject
New-AzNetworkCloudEndpointDependencyObject
New-AzNetworkCloudInitialAgentPoolConfigurationOb…
New-AzNetworkCloudIpAddressPoolObject
New-AzNetworkCloudKeySetUserObject
New-AzNetworkCloudKubernetesCluster
New-AzNetworkCloudKubernetesClusterFeature
New-AzNetworkCloudL2Network
New-AzNetworkCloudL3Network
New-AzNetworkCloudL3NetworkAttachmentConfiguratio…
New-AzNetworkCloudMetricsConfiguration
New-AzNetworkCloudNetworkAttachmentObject
New-AzNetworkCloudRackDefinitionObject
New-AzNetworkCloudServiceLoadBalancerBgpPeerObject
New-AzNetworkCloudServicesNetwork
New-AzNetworkCloudStorageApplianceConfigurationDa…
New-AzNetworkCloudTrunkedNetwork
New-AzNetworkCloudVirtualMachine
New-AzNetworkCloudVirtualMachinePlacementHintObje…
New-AzNetworkCloudVolume
Remove-AzNetworkCloudAgentPool
Remove-AzNetworkCloudBareMetalMachineKeySet
Remove-AzNetworkCloudBmcKeySet
Remove-AzNetworkCloudCluster
Remove-AzNetworkCloudClusterManager
Remove-AzNetworkCloudConsole
Remove-AzNetworkCloudKubernetesCluster
Remove-AzNetworkCloudKubernetesClusterFeature
Remove-AzNetworkCloudL2Network
Remove-AzNetworkCloudL3Network
Remove-AzNetworkCloudMetricsConfiguration
Remove-AzNetworkCloudServicesNetwork
Remove-AzNetworkCloudTrunkedNetwork
Remove-AzNetworkCloudVirtualMachine
Remove-AzNetworkCloudVolume
Restart-AzNetworkCloudBareMetalMachine
Restart-AzNetworkCloudKubernetesClusterNode
Restart-AzNetworkCloudVirtualMachine
Start-AzNetworkCloudBareMetalMachine
Start-AzNetworkCloudVirtualMachine
Stop-AzNetworkCloudBareMetalMachine
Stop-AzNetworkCloudVirtualMachine
Update-AzNetworkCloudAgentPool
Update-AzNetworkCloudBareMetalMachine
Update-AzNetworkCloudBareMetalMachineKeySet
Update-AzNetworkCloudBmcKeySet
Update-AzNetworkCloudCluster
Update-AzNetworkCloudClusterManager
Update-AzNetworkCloudConsole
Update-AzNetworkCloudKubernetesCluster
Update-AzNetworkCloudKubernetesClusterFeature
Update-AzNetworkCloudL2Network
Update-AzNetworkCloudL3Network
Update-AzNetworkCloudMetricsConfiguration
Update-AzNetworkCloudRack
Update-AzNetworkCloudServicesNetwork
Update-AzNetworkCloudStorageAppliance
Update-AzNetworkCloudTrunkedNetwork
Update-AzNetworkCloudVirtualMachine
Update-AzNetworkCloudVolume

### CODE REVIEW

**Comment by VeryEarly** [comment]
please add examples for new cmdlets

File: src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudClusterContinueVersionUpdate.md [hunk 1]
```diff
@@ -48,25 +48,10 @@ Trigger the continuation of an update for a cluster with a matching update strat
 
 ### Example 1: {{ Add title here }}
```
Comment (mbashtovaya) [inline src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudClusterContinueVersionUpdate.md:49]: missing a title

File: src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudScanClusterRuntime.md [hunk 2]
```diff
@@ -47,25 +47,10 @@ Triggers the execution of a runtime protection scan to detect and remediate dete
 
 ### Example 1: {{ Add title here }}
```
Comment (mbashtovaya) [inline src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudScanClusterRuntime.md:48]: here as well

File: src/NetworkCloud/NetworkCloud.Autorest/docs/Remove-AzNetworkCloudKubernetesClusterFeature.md [hunk 3]
```diff
@@ -32,25 +32,10 @@ Delete the provided Kubernetes cluster feature.
 
 ### Example 1: {{ Add title here }}
```
Comment (mbashtovaya) [inline src/NetworkCloud/NetworkCloud.Autorest/docs/Remove-AzNetworkCloudKubernetesClusterFeature.md:33]: and here

**Review by mbashtovaya — CHANGES_REQUESTED** [review]

File: src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudScanClusterRuntime.md (hunk 2 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/docs/Invoke-AzNetworkCloudScanClusterRuntime.md]: updated

**Comment by priyamshet** [comment]
@VeryEarly can we request a review please ?

File: src/NetworkCloud/NetworkCloud.Autorest/docs/Update-AzNetworkCloudKubernetesClusterFeature.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+Update-AzNetworkCloudKubernetesClusterFeature -FeatureName featureName -KubernetesClusterName kubernetesClusterName -ResourceGroupName resourceGroup -SubscriptionId subscriptionId -Tag $tagUpdatedHash
+```
+
+```output
+Location  Name           SystemDataCreatedAt SystemDataCreatedBy    SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy
+--------  ----          -------------------  -------------------    ----------------------- ------------------------  ------------
+uksouth   featureName   12/02/2024 17:44:02	 <identity>             User                     12/02/2024 17:46:45      <identity>
+```
+
+```
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/docs/Update-AzNetworkCloudKubernetesClusterFeature.md:45]: please remove this line

File: src/NetworkCloud/NetworkCloud.Autorest/examples/Invoke-AzNetworkCloudScanClusterRuntime.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+### Example 1: Trigger the execution of a runtime protection scan to detect and remediate detected issues, in accordance with the cluster configuration
+```powershell
+Invoke-AzNetworkCloudScanClusterRuntime -ResourceGroupName resourceGroupName -ClusterName clusterName -SubscriptionId subscriptionId --ScanActivity "Scan"
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/examples/Invoke-AzNetworkCloudScanClusterRuntime.md:3]:
```suggestion
Invoke-AzNetworkCloudScanClusterRuntime -ResourceGroupName resourceGroupName -ClusterName clusterName -SubscriptionId subscriptionId -ScanActivity "Scan"  
```

File: src/NetworkCloud/NetworkCloud.Autorest/examples/Remove-AzNetworkCloudKubernetesClusterFeature.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+### Example 1: Delete the provided Kubernetes cluster feature.
+```powershell
+Remove-AzNetworkCloudKubernetesClusterFeature -FeatureName featureName -KubernetesClusterName kubernetesClusterName -ResourceGroupName resourceGroup -Subscription subscriptionId
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/examples/Remove-AzNetworkCloudKubernetesClusterFeature.md:3]:
```suggestion
Remove-AzNetworkCloudKubernetesClusterFeature -FeatureName featureName -KubernetesClusterName kubernetesClusterName -ResourceGroupName resourceGroup -SubscriptionId subscriptionId 
```

**Comment by priyamshet** [comment]
The changes suggested were incorporated, but we still see the pipeline fail. Is there anyway we can re-trigger Azure pipelines to validate the changes are good?

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudBareMetalMachineConfigurationDataObject.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ### Example 1: Create an in-memory object for BareMetalMachineConfigurationData.
 ```powershell
-$password = ConvertTo-SecureString -String "****" -AsPlainText -Force
+$password = ConvertTo-SecureString "1qaz@WSX" -AsPlainText
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudBareMetalMachineConfigurationDataObject.md:3]: please mask password

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudRackDefinitionObject.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 ### Example 1: Create an in-memory object for RackDefinition.
 ```powershell
-$password = ConvertTo-SecureString -String "****" -AsPlainText -Force
+$password = ConvertTo-SecureString "1qaz@WSX" -AsPlainText
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudRackDefinitionObject.md:3]: please mask password

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudStorageApplianceConfigurationDataObject.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
 ### Example 1: Create an in-memory object for StorageApplianceConfigurationData.
 ```powershell
-$password = ConvertTo-SecureString -String "****" -AsPlainText -Force
+$password = ConvertTo-SecureString "1qaz@WSX" -AsPlainText
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudStorageApplianceConfigurationDataObject.md:3]: please mask password

File: src/NetworkCloud/NetworkCloud.Autorest/test/New-AzNetworkCloudKubernetesCluster.Tests.ps1 [hunk 10]
```diff
@@ -63,7 +62,7 @@ Describe 'New-AzNetworkCloudKubernetesCluster' {
 
             New-AzNetworkCloudKubernetesCluster -ResourceGroupName $kubernetesClusterConfig.resourceGroup `
                 -KubernetesClusterName $kubernetesClusterConfig.kubernetesClusterName -Location  $common.location `
-                -ExtendedLocationName $common.extendedLocation `
+                -ExtendedLocationName $kubernetesClusterConfig.extendedLocation `
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/test/New-AzNetworkCloudKubernetesCluster.Tests.ps1:65]: please mask password at line 21

File: src/NetworkCloud/NetworkCloud.Autorest/test/test-artifacts/config.json [hunk 11]
```diff
@@ -154,59 +160,33 @@
...
   "AzNetworkCloudBareMetalMachine": {
     "bmmName": "rack1compute02",
-    "bmmRg": "simulator-HostedResources-1493033C",
+    "bmmRg": "simdev-2681822-cluster-1-HostedResources-51755DE4",
     "subscriptionId": "a3eeb848-665a-4dbf-80a4-eb460930fb23",
-    "extendedLocation": "/subscriptions/a3eeb848-665a-4dbf-80a4-eb460930fb23/resourcegroups/imulator-HostedResources-1493033C/providers/microsoft.extendedlocation/customlocations/simulator-cstm-loc",
+    "extendedLocation": "/subscriptions/a3eeb848-665a-4dbf-80a4-eb460930fb23/resourceGroups/simdev-2681822-cluster-1-HostedResources-51755DE4/providers/Microsoft.ExtendedLocation/customLocations/simdev-2681822-cluster-1-cstm-loc",
     "tags": "tag1",
     "cordonEvacuate": "False",
     "bmcCredsPassword": "Y0dGemMzZHZjbVFL",
```
Comment (VeryEarly) [inline src/NetworkCloud/NetworkCloud.Autorest/test/test-artifacts/config.json:185]: please mask password

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudStorageApplianceConfigurationDataObject.md (hunk 9 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudStorageApplianceConfigurationDataObject.md]: masked

File: src/NetworkCloud/NetworkCloud.Autorest/test/New-AzNetworkCloudKubernetesCluster.Tests.ps1 (hunk 10 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/test/New-AzNetworkCloudKubernetesCluster.Tests.ps1:65]: updated

File: src/NetworkCloud/NetworkCloud.Autorest/test/test-artifacts/config.json (hunk 11 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/test/test-artifacts/config.json]: UPdated

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudBareMetalMachineConfigurationDataObject.md (hunk 7 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudBareMetalMachineConfigurationDataObject.md]: Updated

File: src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudRackDefinitionObject.md (hunk 8 above)
Comment (priyamshet) [reply src/NetworkCloud/NetworkCloud.Autorest/examples/New-AzNetworkCloudRackDefinitionObject.md]: UPdated

**Review by VeryEarly — APPROVED** [review]

**Comment by priyamshet** [comment]
Details of changes in this PR : 


- This is the stable version of the Az.NetworkCloud that supports NetworkCloud 2024-07-01 APIs.
- Fixed secrets exposure in test recordings, examples and documents

**Pull request merged**